### ISSUE TITLE
Import error for FriedmanHStatisticMethod with version 0.1.4

### ISSUE THREAD

**Issue by conorosully** [pyartemis/artemis#44]
I am using artemis version 0.1.4. I get an error when trying to import the FriedmanHStatisticMethod class using this code: 

from artemis.interactions_methods.model_agnostic import FriedmanHStatisticMethod

See the error: 

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[c:\Users\conorosu\OneDrive](file:///C:/Users/conorosu/OneDrive) - Intel Corporation\Documents\code\h-stat.ipynb Cell 3 line 1
----> [1](vscode-notebook-cell:/c%3A/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/code/h-stat.ipynb#W2sZmlsZQ%3D%3D?line=0) from artemis.interactions_methods.model_agnostic import FriedmanHStatisticMethod

File [c:\Users\conorosu\OneDrive](file:///C:/Users/conorosu/OneDrive) - Intel Corporation\Documents\.venv\Lib\site-packages\artemis\interactions_methods\model_agnostic\__init__.py:1
----> [1](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/__init__.py:1) from .partial_dependence_based import FriedmanHStatisticMethod, GreenwellMethod
      [2](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/__init__.py:2) from .performance_based import SejongOhMethod
      [4](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/__init__.py:4) __all__ = ["FriedmanHStatisticMethod", "GreenwellMethod", "SejongOhMethod"]

File [c:\Users\conorosu\OneDrive](file:///C:/Users/conorosu/OneDrive) - Intel Corporation\Documents\.venv\Lib\site-packages\artemis\interactions_methods\model_agnostic\partial_dependence_based\__init__.py:1
----> [1](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/__init__.py:1) from ._greenwell import GreenwellMethod
      [2](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/__init__.py:2) from ._friedman_h_statistic import FriedmanHStatisticMethod
      [4](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/__init__.py:4) __all__ = ["FriedmanHStatisticMethod", "GreenwellMethod"]

File [c:\Users\conorosu\OneDrive](file:///C:/Users/conorosu/OneDrive) - Intel Corporation\Documents\.venv\Lib\site-packages\artemis\interactions_methods\model_agnostic\partial_dependence_based\_greenwell.py:11
      [9](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:9) from artemis._utilities.ops import partial_dependence_value, split_features_num_cat
     [10](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:10) from artemis._utilities.pd_calculator import PartialDependenceCalculator
---> [11](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:11) from ._pdp import PartialDependenceBasedMethod
     [14](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:14) class GreenwellMethod(PartialDependenceBasedMethod):
     [15](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:15)     """
     [16](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:16)     Greenwell Method for Feature Interaction Extraction. 
     [17](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:17)     
ref='c:\Users\conorosu\OneDrive - Intel Corporation\Documents\.venv\Lib\site-packages\artemis\interactions_methods\model_agnostic\partial_dependence_based\_greenwell.py:1'>1</a>;32m   (...)
     [45](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:45)     - https://arxiv.org/pdf/1805.04755.pdf
     [46](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_greenwell.py:46)     """

File [c:\Users\conorosu\OneDrive](file:///C:/Users/conorosu/OneDrive) - Intel Corporation\Documents\.venv\Lib\site-packages\artemis\interactions_methods\model_agnostic\partial_dependence_based\_pdp.py:11
      [8](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_pdp.py:8) import seaborn as sns
     [10](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_pdp.py:10) from artemis.importance_methods.model_agnostic import PartialDependenceBasedImportance
---> [11](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_pdp.py:11) from artemis.interactions_methods._method import FeatureInteractionMethod
     [12](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_pdp.py:12) from artemis._utilities.domain import ProgressInfoLog, VisualizationType
     [13](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/model_agnostic/partial_dependence_based/_pdp.py:13) from artemis._utilities.ops import get_predict_function, sample_if_not_none, all_if_none

File [c:\Users\conorosu\OneDrive](file:///C:/Users/conorosu/OneDrive) - Intel Corporation\Documents\.venv\Lib\site-packages\artemis\interactions_methods\_method.py:9
      [7](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/_method.py:7) from artemis._utilities.domain import VisualizationType
      [8](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/_method.py:8) from artemis._utilities.exceptions import MethodNotFittedException
----> [9](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/_method.py:9) from artemis.visualizer._configuration import VisualizationConfigurationProvider
     [10](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/_method.py:10) from artemis.visualizer._visualizer import Visualizer
     [13](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/interactions_methods/_method.py:13) class FeatureInteractionMethod(ABC):

File [c:\Users\conorosu\OneDrive](file:///C:/Users/conorosu/OneDrive) - Intel Corporation\Documents\.venv\Lib\site-packages\artemis\visualizer\_configuration.py:176
    [166](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:166)         graph_config.MAX_EDGE_WIDTH = 3
    [168](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:168)         return VisualizationConfiguration(
    [169](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:169)             accepted_visualizations=cls.accepted_visualizations[
    [170](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:170)                 InteractionMethod.CONDITIONAL_MINIMAL_DEPTH
    [171](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:171)             ],
    [172](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:172)             interaction_graph=graph_config,
    [173](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:173)         )
--> [176](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:176) @dataclass
    [177](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:177) class VisualizationConfiguration:
    [178](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:178)     accepted_visualizations: List[str]
    [179](file:///C:/Users/conorosu/OneDrive%20-%20Intel%20Corporation/Documents/.venv/Lib/site-packages/artemis/visualizer/_configuration.py:179)     interaction_graph: InteractionGraphConfiguration = InteractionGraphConfiguration()

File [C:\Program](file:///C:/Program) Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0\Lib\dataclasses.py:1230, in dataclass(cls, init, repr, eq, order, unsafe_hash, frozen, match_args, kw_only, slots, weakref_slot)
   [1227](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:1227)     return wrap
   [1229](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:1229) # We're called as @dataclass without parens.
-> [1230](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:1230) return wrap(cls)

File [C:\Program](file:///C:/Program) Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0\Lib\dataclasses.py:1220, in dataclass.<locals>.wrap(cls)
   [1219](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:1219) def wrap(cls):
-> [1220](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:1220)     return _process_class(cls, init, repr, eq, order, unsafe_hash,
   [1221](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:1221)                           frozen, match_args, kw_only, slots,
   [1222](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:1222)                           weakref_slot)

File [C:\Program](file:///C:/Program) Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0\Lib\dataclasses.py:958, in _process_class(cls, init, repr, eq, order, unsafe_hash, frozen, match_args, kw_only, slots, weakref_slot)
    [955](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:955)         kw_only = True
    [956](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:956)     else:
    [957](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:957)         # Otherwise it's a field of some type.
--> [958](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:958)         cls_fields.append(_get_field(cls, name, type, kw_only))
    [960](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:960) for f in cls_fields:
    [961](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:961)     fields[f.name] = f

File [C:\Program](file:///C:/Program) Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0\Lib\dataclasses.py:815, in _get_field(cls, a_name, a_type, default_kw_only)
    [811](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:811) # For real fields, disallow mutable defaults.  Use unhashable as a proxy
    [812](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:812) # indicator for mutability.  Read the __hash__ attribute from the class,
    [813](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:813) # not the instance.
    [814](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:814) if f._field_type is _FIELD and f.default.__class__.__hash__ is None:
--> [815](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:815)     raise ValueError(f'mutable default {type(f.default)} for field '
    [816](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:816)                      f'{f.name} is not allowed: use default_factory')
    [818](file:///C:/Program%20Files/WindowsApps/PythonSoftwareFoundation.Python.3.11_3.11.1776.0_x64__qbz5n2kfra8p0/Lib/dataclasses.py:818) return f

ValueError: mutable default <class 'artemis.visualizer._configuration.InteractionGraphConfiguration'> for field interaction_graph is not allowed: use default_factory

**Comment by conorosully** [comment]
Following up with the above issue. I downgraded to version 0.1.3 but still have the same issue. I also tried to import other classes (e.g. GreenwellMethod) but get a similar error:

ValueError: mutable default <class 'artemis.visualizer._configuration.InteractionGraphConfiguration'> for field interaction_graph is not allowed: use default_factory

I hope you guys can fix it. This package looks very useful! Not much out there to analyze interactions with Python. 

All the best!
